### PR TITLE
fix: remove p-settle dependency, use Promise.allSettled instead

### PR DESCRIPTION
**Pull request by jimmywarting**
remove dependency on `p-settle` and use native, spec'ed / builtin `Promise.allSettled` instead.

### CODE REVIEW

**Comment by achingbrain** [comment]
`p-settle` is used in the [transport manager](https://github.com/jimmywarting/js-libp2p/blob/rm-pSettle/src/transport-manager.ts#L2) so this will need refactoring too

**Comment by jimmywarting** [comment]
fixed.

File: test/upgrading/upgrader.spec.ts [hunk 1]
```diff
@@ -255,15 +254,15 @@ describe('Upgrader', () => {
...
+    const results = await Promise.allSettled([
       localUpgrader.upgradeOutbound(outbound),
       remoteUpgrader.upgradeInbound(inbound)
     ])
 
     // Ensure both sides fail
     expect(results).to.have.length(2)
     results.forEach(result => {
-      expect(result).to.have.property('isRejected', true)
+      expect(result.status).to.equal('rejected')
```
Comment (achingbrain) [inline test/upgrading/upgrader.spec.ts:265]:
```suggestion
      expect(result).to.have.property('status', 'rejected')
```

File: test/upgrading/upgrader.spec.ts [hunk 2]
```diff
@@ -311,15 +310,15 @@ describe('Upgrader', () => {
...
```
Comment (achingbrain) [inline test/upgrading/upgrader.spec.ts:321]:
```suggestion
      expect(result).to.have.property('status', 'rejected')
```

File: test/upgrading/upgrader.spec.ts [hunk 3]
```diff
@@ -401,13 +400,13 @@ describe('Upgrader', () => {
...
 
-    const results = await pSettle([
+    const results = await Promise.allSettled([
       connections[0].newStream('/unsupported/1.0.0'),
       connections[1].newStream('/unsupported/1.0.0')
     ])
     expect(results).to.have.length(2)
     results.forEach(result => {
-      expect(result).to.have.property('isRejected', true)
+      expect(result.status).to.equal('rejected')
```
Comment (achingbrain) [inline test/upgrading/upgrader.spec.ts:409]:
```suggestion
      expect(result).to.have.property('status', 'rejected')
```

**Review by achingbrain — APPROVED** [review]

**Comment by achingbrain** [comment]
Thanks!

**Pull request merged**